tests/runner.py

```diff
@@ -184,18 +185,24 @@ def uses_memory_init_file(self):
       # side modules handle memory differently; binaryen puts the memory in the wasm module
       return ('-O2' in self.emcc_args or '-O3' in self.emcc_args or '-Oz' in self.emcc_args) and not (Settings.SIDE_MODULE or Settings.BINARYEN)
 
+  def canonical_temp_dir(self):
+    return os.path.join(self.temp_dir, 'emscripten_temp')
```

Comment:
we still have logic in `shared.py` that uses that string "emscripten_temp", can we share that code so it's not duplicated?